language/move-lang/src/expansion/translate.rs

```diff
@@ -214,57 +256,65 @@ fn set_sender_address(context: &mut Context, loc: Loc, sender: Option<Address>)
     })
 }
 
-fn module_(
-    context: &mut Context,
-    is_source_module: bool,
-    mdef: P::ModuleDefinition,
-) -> (ModuleIdent, E::ModuleDefinition) {
-    assert!(context.aliases.is_empty());
-    let P::ModuleDefinition {
-        loc,
-        uses,
-        name,
-        structs: mut pstructs,
-        functions: mut pfunctions,
-        specs: pspecs,
-    } = mdef;
+fn module_(context: &mut Context, mdef: P::ModuleDefinition) -> (ModuleIdent, E::ModuleDefinition) {
+    let P::ModuleDefinition { loc, name, members } = mdef;
     let name_loc = name.loc();
     if name.value() == ModuleName::SELF_NAME {
         context.restricted_self_error("module", &name.0);
     }
 
     let mident_ = ModuleIdent_ {
         address: context.cur_address(),
-        name: name.clone(),
+        name,
     };
     let current_module = ModuleIdent(sp(name_loc, mident_));
-    let is_source_module = is_source_module && !fake_natives::is_fake_native(&current_module);
-    let self_aliases = module_self_aliases(&current_module);
-    context.set_and_shadow_aliases(self_aliases);
-    let alias_map = aliases(context, uses);
-    context.set_and_shadow_aliases(alias_map.clone());
-    if !is_source_module {
-        for pstruct in pstructs.iter_mut() {
-            pstruct.fields = P::StructFields::Native(pstruct.loc)
-        }
-        for pfunction in pfunctions.iter_mut() {
-            pfunction.body.value = P::FunctionBody_::Native
+    let old_is_source_module = context.is_source_module;
+    context.is_source_module =
+        context.is_source_module && !fake_natives::is_fake_native(&current_module);
+
+    let mut new_scope = AliasMap::new();
+    module_self_aliases(&mut new_scope, &current_module);
+    let members = members
+        .into_iter()
+        .filter_map(|member| aliases_from_member(context, &mut new_scope, &current_module, member))
+        .collect::<Vec<_>>();
```

Comment:
Since you're going to be iterating over the members later you can probably elide this `collect`